sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
+                b = base.__class__(Identity(base.shape[0]))
+                if b == Identity(base.shape[0]).as_explicit():
+                    return b
+                else:
+                    return Identity(base.shape[0])
+            except TypeError:
```

Comment:
`is_explicit` would tell you if a Matrix is explicit or not (i.e., if it is a matrix expression or a Matrix). is_identity would tell you if a matrix is an identity matrix. Both would be implemented on MatrixBase and MatrixExpr. 

I guess instead of `is_explicit` you could just use `isinstance(mat, MatrixBase)`.